src/System.Private.CoreLib/src/System/StartupHookProvider.cs

```diff
@@ -31,34 +34,52 @@ private static void ProcessStartupHooks()
             }
 
             // Parse startup hooks variable
-            string[] startupHooks = startupHooksVariable.Split(Path.PathSeparator);
-            foreach (string startupHook in startupHooks)
+            string[] startupHookParts = startupHooksVariable.Split(Path.PathSeparator);
+            StartupHookNameOrPath[] startupHooks = new StartupHookNameOrPath[startupHookParts.Length];
+            for (int i = 0; i < startupHookParts.Length; i++)
             {
-                if (String.IsNullOrEmpty(startupHook))
+                string startupHookPart = startupHookParts[i];
+                if (string.IsNullOrEmpty(startupHookPart))
```

Comment:
Nit: I am wondering whether we should skip empty parts here instead of throwing to make it easier for folks building string with multiple hooks - they will be able to always append PathSeparator.

Reply:
I'll let @sbomer comment on that. The current code is very strict, fails on trailing path separator as well.

Reply:
To follow up, if the method `ProcessStartupHooks` throws, does that cause startup to fail?

Reply:
Yes

Reply:
Yes, it's currently pretty strict, but I have no objections to relaxing this to make it easier to use.

Reply:
Any plans to fix this? (It is 3 line change...)

Reply:
Sorry - added this functionality in the most recent commits.
It now allows extra path separators. It still doesn't allow whitespaces, I considered it, but it felt wrong.